File: 2024-03-12_Robot_path/c/robot.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main(int argc, char *argv[]) {
    if (argc != 2) {
        printf("Usage: %s <sequence>\n", argv[0]);
        return 1;
    }

    char *sequence = argv[1];

    printf("%s\n", sequence);

    int x = 0;
    int y = 0;

    srand(time(NULL));

    for (int i = 0; i < 1000; i++) {
        printf("Robot is moving randomly\n");
        char xory = (rand() > RAND_MAX / 2) ? 'x' : 'y';
        if (xory == 'x') {
            x += (rand() > RAND_MAX / 2) ? 1 : -1;
        } else {
            y += (rand() > RAND_MAX / 2) ? 1 : -1;
        }

        printf("> %d;%d\n", x, y);
    }

    return 0;
}

File: 2024-03-12_Robot_path/robot.py
import random
import sys

sequence = sys.argv[1]

print(sequence);

x = 0
y = 0

for i in range(1000):
    print("Robot is moving randomly")
    xory = "x" if random.random() > 0.5 else "y"
    if xory == "x":
        x += 1 if random.random() > 0.5 else -1
    else:
        y += 1 if random.random() > 0.5 else -1

    print(f"> {x};{y}")
